crates/red_knot_python_semantic/src/types/call/bind.rs

```diff
@@ -26,29 +29,79 @@ use ruff_text_size::Ranged;
 /// compatible with _all_ of the types in the union for the call to be valid.
 ///
 /// It's guaranteed that the wrapped bindings have no errors.
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(Debug)]
 pub(crate) struct Bindings<'db> {
-    pub(crate) callable_type: Type<'db>,
+    signatures: Signatures<'db>,
     /// By using `SmallVec`, we avoid an extra heap allocation for the common case of a non-union
     /// type.
     elements: SmallVec<[CallableBinding<'db>; 1]>,
+
+    /// Whether each argument will be used as a value and/or a type form in this call.
+    ///
+    /// TODO: We will eventually infer completely different argument types for each signature, once
+    /// we are able to use the annotated parameter types as type contexts for that inference. At
+    /// that point, this field will move down into `CallBinding` or `Binding`.
```

Comment:
This TODO may not be accurate, given our conversation today? Totally up to you how you want to handle this.

Reply:
Removed